src/main/java/org/broadinstitute/consent/http/service/ElasticSearchService.java

```diff
@@ -278,7 +278,12 @@ public Response indexDataset(Dataset dataset, User user) throws IOException {
   }
 
   public Response indexDatasets(List<Dataset> datasets, User user) throws IOException {
-    List<DatasetTerm> datasetTerms = datasets.stream().map(this::toDatasetTerm).toList();
+    // Datasets in list context may not have their study populated, so we need to ensure that is
```

Comment:
If this code is only using the dataset IDs, I would change it to accept `List<Integer> datasetIds` instead, to make this clear. For example, `ElasticSearchService.indexStudy()` is calling this method after also calling `findDatasetsByIdList()`, resulting in duplicate dataset lookups.

Reply:
Yes, I think this will simplify some other usages as well 👍🏽